From: Amon Ott <ao@m-privacy.de>
Date: Mon, 23 Jan 2012 09:25:23 -0800
Subject: [PATCH 090/199] ceph: use 2 instead of 1 as fallback for 32-bit inode number

Git-commit: a661fc561190c0ee2d7cfabcfa92204e2b3aa349
Patch-mainline: v3.4
References: ceph backport

The root directory of the Ceph mount has inode number 1, so falling back
to 1 always creates a collision. 2 is unused on my test systems and seems
less likely to collide.

Signed-off-by: Amon Ott <ao@m-privacy.de>
Signed-off-by: Sage Weil <sage@newdream.net>
Signed-off-by: Guangliang Zhao <gzhao@suse.com>
---
 fs/ceph/super.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/fs/ceph/super.h b/fs/ceph/super.h
index 1421f3d..18d8a86 100644
--- a/fs/ceph/super.h
+++ b/fs/ceph/super.h
@@ -367,7 +367,7 @@ static inline u32 ceph_ino_to_ino32(__u64 vino)
 	u32 ino = vino & 0xffffffff;
 	ino ^= vino >> 32;
 	if (!ino)
-		ino = 1;
+		ino = 2;
 	return ino;
 }
 
-- 
1.7.3.4

